components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -215,7 +215,17 @@ public static String getAuthenticatedUsername() {
      */
     public static String getUserId() {
 
-        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+        String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+        if (userId == null && PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername() != null) {
+            try {
+                userId = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
```

Comment:
Add a comment by describing requirement to solve user id in this way

Reply:
Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/15/commits/5c2ff91a136fe4042e3754d26718c8ca3a1b12a3